src/ddmd/mtype.d

```diff
@@ -3058,16 +3058,23 @@ extern (C++) abstract class Type : RootObject
      * https://issues.dlang.org/show_bug.cgi?id=14488
      * Check if the inner most base type is complex or imaginary.
      * Should only give alerts when set to emit transitional messages.
+     * Params:
+     *  loc = The source location.
      */
     final void checkComplexTransition(Loc loc)
     {
         Type t = baseElemOf();
         while (t.ty == Tpointer || t.ty == Tarray)
             t = t.nextOf().baseElemOf();
+        if (t.ty == Tenum)
+            t = (cast(TypeEnum)t).sym.memtype;
+
+        // Type is forward referenced or not a scalar type.
+        if (t is null || !t.isTypeBasic())
```

Comment:
Is the check for typebasic really necessary? This should be covered by line 3076. Also, the null check seems unnecessary. Is there a test case for this? According to the previous logic, it could only be null if `memtype` is null. Can this ever happen?